src/phpDocumentor/Configuration/ConfigurationFactory.php

```diff
@@ -0,0 +1,332 @@
+        }
+
+        return $array;
+    }
+
+    /**
+     * Validates if the Uri contains an xml that has a root element which name is phpdocumentor.
+     *
+     * @param Uri $uri
+     *
+     * @return \SimpleXMLElement
+     */
+    private function validate(Uri $uri)
+    {
+        $xml = new \SimpleXMLElement($uri, 0, true);
+
+        if ($xml->getName() !== 'phpdocumentor') {
+            throw new \InvalidArgumentException(sprintf('Root element name should be phpdocumentor, %s found', $xml->getName()));
+        }
+
+        return $xml;
+    }
+
+    /**
+     * Checks if version attribute is present. If found, it is phpDocumentor3 configuration.
+     * If no version attribute is found, it is assumed that it is phpDocumentor2 configuration.
+     *
+     * @param \SimpleXMLElement $phpDocumentor
+     *
+     * @return bool
+     */
+    private function checkIfVersionAttributeIsPresent(\SimpleXMLElement $phpDocumentor)
+    {
+        return isset($phpDocumentor->attributes()->version);
+    }
+
+    /**
+     * Converts the phpDocumentor2 configuration xml to an array
+     *
+     * @param \SimpleXMLElement $phpDocumentor
+     *
+     * @return array
+     */
+    private function convertPhpdoc2XmlToArray(\SimpleXMLElement $phpDocumentor)
+    {
+        $extensions         = [];
+        $markers            = [];
+        $visibility         = 'public';
+        $defaultPackageName = 'Default';
+        $template           = 'clean';
+        $ignoreHidden       = true;
+        $ignoreSymlinks     = true;
+
+        if (isset($phpDocumentor->parser)) {
+            $extensions = $this->buildExtensionsPart($phpDocumentor->parser);
+            $markers    = $this->buildMarkersPart($phpDocumentor->parser);
+
+            $visibility         = ((string) $phpDocumentor->parser->visibility) ?: $visibility;
+            $defaultPackageName = ((string) $phpDocumentor->parser->{'default-package-name'}) ?: $defaultPackageName;
+            $template           = ((string) $phpDocumentor->transformations->template->attributes()->name) ?: $template;
+
+            if (isset($phpDocumentor->parser->files)) {
+                if (isset($phpDocumentor->parser->files->{'ignore-hidden'})) {
+                    $ignoreHidden = filter_var($phpDocumentor->parser->files->{'ignore-hidden'}, FILTER_VALIDATE_BOOLEAN);
+                }
+
+                if (isset($phpDocumentor->parser->files->{'ignore-symlinks'})) {
+                    $ignoreSymlinks = filter_var($phpDocumentor->parser->files->{'ignore-symlinks'}, FILTER_VALIDATE_BOOLEAN);
+                }
+            }
+        }
+
+        $phpdoc2Array = [
+            'phpdocumentor' => [
+                'paths'     => [
+                    'output' => 'file://build/docs',
+                    'cache'  => '/tmp/phpdoc-doc-cache'
+                ],
+                'versions'  => [
+                    '1.0.0' => [
+                        'folder' => 'latest',
```

Comment:
Can you make this empty? The default output (sub)folder for the version 2 xml is `''` (otherwise we'd break BC a bit too much ;))